mvi-presenter/src/main/java/pl/valueadd/mvi/presenter/BaseMviPresenter.kt

```diff
@@ -45,9 +45,8 @@ abstract class BaseMviPresenter<VS : IBaseViewState, PS : IBasePartialState, VI
      * @see subscribeViewStateConsumer
      * @see viewStateConsumerDisposable
      */
-    protected open val viewStateObservationScheduler: Scheduler by lazy {
-        AndroidSchedulers.mainThread()
-    }
+    protected open val viewStateObservationScheduler: Scheduler =
```

Comment:
You can change this to 
`abstract class BaseMviPresenter<VS : IBaseViewState, PS : IBasePartialState, VI : IBaseView.IBaseIntent, V : IBaseView<VS, VI>>(
    protected open val viewStateObservationScheduler: Scheduler
) : IMviPresenter<V> {`

Reply:
It would be the same but now it looks clearly with other fields and documentation